src/main/webapp/app/exercises/shared/exercise/exercise.service.ts

```diff
@@ -119,7 +119,7 @@ export class ExerciseService {
      * @param { number } exerciseId - Id of exercise that should be resetted
      */
     reset(exerciseId: number): Observable<HttpResponse<void>> {
-        return this.http.delete<void>(`${this.resourceUrl}/${exerciseId}/reset`, { observe: 'response' });
+        return this.http.post<void>(`${this.resourceUrl}/${exerciseId}/reset`, undefined, { observe: 'response' });
```

Comment:
Just an observation: we don't seem to have any guidelines whether we pass `undefined` or `{}` in such cases. Since we use both (but `{}` more often), this should be fine, but maybe someone knows what is better here, and we could maybe add something to the guidelines.